picodom.d.ts

```diff
@@ -0,0 +1,30 @@
+declare namespace picodom {
+  export interface VNode<D> {
+    tag: string
+    data?: D
+    children: VNode<{}>
+  }
+
+  export interface Component<D> {
+    (data?: D, ...children: VNode<{}>[]): VNode<D>
+  }
+
+  export type VElement<D> = Component<D> | string
```

Comment:
@pspeter3 Can we somehow avoid declaring VElement and just put it in the definition of `tag` below?